Upgrade Slate and implement new schema

Zhouzi opened the pull request:
This PR upgrades Slate to the latest version and updates the schema to use the new definition.

Soreine commented on lib/validation/validateNode.js:
```diff
@@ -1,245 +1,35 @@
-
-        validate(node) {
-            const { nodes } = node;
+        if (!opts.allowMarks && node.type === opts.lineType) {
```
Can't you use the marks validation from the new schema ?

https://docs.slatejs.org/other-packages/slate-schema-violations#node_mark_invalid

Zhouzi commented:
@Soreine thanks, I thought it was not possible, it's now fixed :) I updated Slate's documentation to mention that option (it was missing): https://github.com/ianstormtaylor/slate/pull/1717

Soreine commented on lib/validation/schema.js:
```diff
@@ -41,6 +41,12 @@ function schema(opts: Options): Object {
     };
+
+    if (!opts.allowMarks) {
+        baseSchema.blocks[opts.lineType].marks = [];
```
Much more concise <3

Soreine commented:
You rock ❤️
